prow/tekton.md

```diff
@@ -0,0 +1,147 @@
+
+### `pipeline` and `crier` controllers
+
+In addition to the services required by `Prow` for `prowjobs`, you will need
+to add:
+
+1. [The pipeline controller](/prow/cmd/pipeline): Creates
+   [`PipelineRuns](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/pipelineruns.md)
+2. [The crier controller](/prow/cmd/crier): Updates GitHub with the results of
+   `ProwJobs` executed by agent types other than `kubernetes`
+
+You can add these to your cluster with:
+
+```bash
+# Add the service account for the crier controller
+k apply -f prow/cluster/crier_rbac.yaml
+
+# Add the service account for the pipeline controller
+k apply -f prow/cluster/pipeline_rbac.yaml
+
+# Add the pipeline controller deployment
+k apply -f prow/cluster/simple_pipeline_deployment.yaml
+
+# Add the crier controller deployment
+k apply -f prow/cluster/simple_crier_deployment.yaml
+
+```
+
+## Configuring a `tekton-pipeline` job
+
+Once [you have everything setup](#setting-up-pipelines-with-prow) you can
+configure Prow jobs to run Pipelines.
+
+1. [Create and apply `Pipelines` and
+   `Tasks`](#create-and-apply-pipelines-and-tasks)
+2. [Configure Prow with `ProwJobs` that use those Pipelines and Tasks](#add-tekton-pipeline-prowjobws)
+3. Then when the `ProwJob` is triggered:
+  a. Prow (via [the Prow pipeline controller](#pipeline-and-crier-controllers))
+     will create [a PipelineRun](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/pipelineruns.md)
+  b. When the `PipelineRun` completes, the Prow Pipeline controller will update
+     the `ProwJob`
+  c. Finally, [crier](#pipeline-and-crier-controllers) will see the update to
+     the `ProwJob` and update `GitHub` with the results
+
+### Create and apply `Pipelines` and `Tasks`
+
+Before anything can be executed, Tekton expects
+[`Pipelines`](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/pipelines.md)
+and the [`Tasks`](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/tasks.md)
+that the `Pipelines` reference to exist.
+
+Since [Prow is only compatible with versions 0.2.0 - 0.3.1 of Tekton Pipelines](https://github.com/kubernetes/test-infra/issues/13948)
+the below docs are pinned to v0.3.1 of Tekton Pipelines:
+
+* See [the Tekton tutorial](https://github.com/tektoncd/pipeline/blob/v0.3.1/docs/tutorial.md)
+  for an overview
+* See [the Tekton Pipelines docs](https://github.com/tektoncd/pipeline/tree/v0.3.1/docs#tekton-pipelines)
+  for reference docs
+* See [the Tekton catalog](https://github.com/tektoncd/catalog)
+  for examples of `Tasks` you might want to use.
+
+### Add `tekton-pipeline` ProwJobs
+
+To configure a `ProwJob` to use Tekton Pipelines, include a `pipelineRunSpec`,
+for example:
+
+```yaml
+presubmits:
+  bobcatfish/catservice:
+  - name: do-the-pipeline
+    agent: tekton-pipeline # Use the Prow Tekton controller!
+    rerun_command: "/run do-the-pipeline"
+    trigger: "(?m)^/run do-the-pipeline,?(\\s+|$)"
+    always_run: true
+    skip_report: false
```

Comment:
nit: I'd omit the `skip_report`, `rerun_command`, and `trigger` fields. They aren't necessary if you are ok with using `/test do-the-pipeline` to trigger the job.

Reply:
kk!